contracts/airdrops/Existing1155NftDrop.sol

```diff
@@ -68,69 +53,46 @@ contract Existing1155NftDrop is AirdropInfo, AirdropMerkleProof, IERC1155Receive
         airdropFunded = true;
         airdropFundBlockTimestamp = block.timestamp;
         airdropFundingHolder = msg.sender;
-        emit AirdropFunded();
+        emit AirdropFunded(address(this));
     }
 
     /// @notice Allows the airdrop creator to withdraw back his funds after the airdrop has finished
     function withdrawAirdropFunds() external {
-        if (airdropFundingHolder != msg.sender) revert NotOwner();
+        if (airdropFundingHolder != msg.sender) revert Unauthorized();
         if (block.timestamp < airdropFinishTime) revert AirdropStillInProgress();
-        rewardToken.safeTransferFrom(
-            address(this),
-            msg.sender,
-            rewardTokenId,
-            rewardToken.balanceOf(address(this), rewardTokenId),
-            ""
-        );
+        rewardToken.safeTransferFrom(address(this), msg.sender, rewardTokenId, rewardToken.balanceOf(address(this), rewardTokenId), "");
     }
 
     /// @notice Allows the NFT holder to claim his ERC20 airdrop
-    function claim(uint256 tokenId, bytes32[] calldata _merkleProof) external {
-        if (hasClaimed[tokenId]) revert AlreadyRedeemed();
-        if (rewardToken.balanceOf(address(this), rewardTokenId) < tokensPerClaim) revert InsufficientLiquidity();
-        if (rewardedNft.ownerOf(tokenId) != msg.sender) revert NotOwner();
-        if (block.timestamp > airdropFinishTime) revert AirdropExpired();
-
-        checkProof(_merkleProof, merkleRoot);
-
-        hasClaimed[tokenId] = true;
-        emit Claimed(tokenId, msg.sender);
-
-        rewardToken.safeTransferFrom(address(this), msg.sender, rewardTokenId, tokensPerClaim, "");
+    function claim(uint256 tokenId) external {
+        if (isEligibleForReward(tokenId)) {
+            hasClaimed[tokenId] = true;
+            rewardToken.safeTransferFrom(address(this), msg.sender, rewardTokenId, tokensPerClaim, "");
+            emit Claimed(tokenId, msg.sender);
+        } else {
+            revert NotEligible();
+        }
     }
 
-    function batchClaim(uint256[] memory tokenIds, bytes32[] calldata _merkleProof) external {
-        if (rewardToken.balanceOf(address(this), rewardTokenId) < tokensPerClaim * tokenIds.length)
-            revert InsufficientLiquidity();
+    function batchClaim(uint256[] memory tokenIds) external {
+        if (rewardToken.balanceOf(address(this), rewardTokenId) < tokensPerClaim * tokenIds.length) revert InsufficientLiquidity();
 
         if (block.timestamp > airdropFinishTime) revert AirdropExpired();
 
-        checkProof(_merkleProof, merkleRoot);
-
         for (uint256 index = 0; index < tokenIds.length; index++) {
             uint256 tokenId = tokenIds[index];
 
             if (hasClaimed[tokenId]) revert AlreadyRedeemed();
-            if (rewardedNft.ownerOf(tokenId) != msg.sender) revert NotOwner();
+            if (rewardedNft.ownerOf(tokenId) != msg.sender) revert Unauthorized();
```

Comment:
see line 108 comment please

Reply:
I'm not sure If understand what you are referring to

Reply:
Forget that I wrote it